src/api/Tron.js

```diff
@@ -275,6 +276,77 @@ export const getContractUserEnergyRatioConsumption = async (
   return consume_user_resource_percent;
 };
 
+export const getAccountDelegatedResourcesDetails = async (
+  fromAddress: string,
+  toAddress: string
+): Promise<DelegatedResources[]> => {
+  const result = await fetch(
+    `${baseApiUrl}/wallet/getdelegatedresource?fromAddress=${fromAddress}&toAddress=${toAddress}`
+  );
+
+  const resources: DelegatedResources[] = [];
+  if (
+    result &&
+    result.delegatedResource &&
+    result.delegatedResource.length === 1
+  ) {
+    const {
+      frozen_balance_for_bandwidth,
+      frozen_balance_for_energy,
+      expire_time_for_bandwidth,
+      expire_time_for_energy
+    } = result.delegatedResource[0];
+
+    if (frozen_balance_for_bandwidth) {
+      resources.push({
+        toAddress: encode58Check(toAddress),
+        resource: "BANDWIDTH",
+        amount: BigNumber(frozen_balance_for_bandwidth),
+        expiredAt: new Date(expire_time_for_bandwidth)
+      });
+    }
+    if (frozen_balance_for_energy) {
+      resources.push({
+        toAddress: encode58Check(toAddress),
+        resource: "ENERGY",
+        amount: BigNumber(frozen_balance_for_energy),
+        expiredAt: new Date(expire_time_for_energy)
+      });
+    }
+  }
+  return resources;
+};
+
+export const getAccountDelegatedResources = async (
+  fromAddress: string
+): Promise<DelegatedResources[]> => {
+  const { toAccounts } = await fetch(
+    `${baseApiUrl}/wallet/getdelegatedresourceaccountindex?value=${fromAddress}`
+  );
+
+  const resources: DelegatedResources[] = [];
+  if (toAccounts) {
+    var promises = [];
+    toAccounts.forEach(toAddress => {
+      promises.push(
+        getAccountDelegatedResourcesDetails(fromAddress, toAddress)
+          .then(resourceToAddress => {
+            resources.push(...resourceToAddress);
+          })
+          .catch(e => {
+            log(
+              "tron-error",
+              "fetch account delegated resources fails with " + e.message,
+              { fromAddress }
+            );
+          })
+      );
+    });
+    await Promise.all(promises);
```

Comment:
prefer usage of our internal `promiseAllBatched` , we had bad surprise when doing too much Promise.all with tron api.

also i'm a bit scared about the performance of this, how much "toAccounts" would there be in usual accounts?

Reply:
Usually 1-2, some may have about 10. Maybe a dApp creator which may need to delegate energy to several contracts...